### PR TITLE
[FOS-11703] Add initial stub implementation.

### PR DESCRIPTION
**Pull request by adamcalhoon**
Still WIP -- I need to make sure the driver actually comes up.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  robot-abstraction-layer
+  robot-abstraction-layer-utils
+  forge-core-logger-cpp
+  forge-core-storage
+  forge-core-common
+  shared-library-cpp
+  ral-messages
+  ral2-driver
+  robot-configuration
+  robot-viz-tools
```
Comment (mechterp10) [inline CMakeLists.txt]: What is robot-viz-tools required for?

File: configurations/Sample.json [hunk 2]
```diff
@@ -0,0 +1,81 @@
+{
+    "device": {
+        "name": "RAL2 Sample Driver"
+    },
+    "attributes": {
+        "robot-driver-library": "libral2-sample-driver.so",
+        "jog-controllers.cartesian.library": "libopen-loop-cartesian-iterative-position.so",
```
Comment (mechterp10) [inline configurations/Sample.json]: Defaulting to the cartesian velocity jogging may make more sense. Also, with a jog frame of TCP.

File: CMakeLists.txt (hunk 1 above)
Comment (adamcalhoon) [reply CMakeLists.txt]: I'm not sure it is. I copied the doosan, which copied the kawasaki which copied the....

File: configurations/controller-models/RCU/Program Mode.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+    "motion-exec-types": ["VEL", "ACC_VEL", "TIME"],
+    "standard-tool-orientation-values": ["ALIGN_TO_END_IGNORE_INTERMEDIATE"],
+    "prefer-cartesian-targets": true,
+    "is-keyless": true,
+    "robot-reduced-is-program": true,
+    "disconnected-fault-override-code": 100000,
```
Comment (mechterp10) [inline configurations/controller-models/RCU/Program Mode.json:7]: I would add this error code in the Sample.json file?

File: src/SampleDriver.cpp [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+#include <QTimer>
+
+static ral::Pose identityPose()
+{
+    ral::Pose result;
+
+    // No translation
+    result.mutable_translation_mm()->add_data(0.0);
+    result.mutable_translation_mm()->add_data(0.0);
+    result.mutable_translation_mm()->add_data(0.0);
```
Comment (mechterp10) [inline src/SampleDriver.cpp]: Nitpick,
```suggestion
    result.mutable_translation_mm()->mutable_data()->Resize(3, 0.0);
```

Comment (mechterp10) [reply src/SampleDriver.cpp]: Applies for most of these

File: src/SampleDriver.cpp [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+    driver_state.mutable_speed()->set_speed_factor(1.0);
+    driver_state.set_safety_override(ral::State::NONE);
+    emit controllerStateChanged(driver_state);
+
+    ral::DriverConfiguration driver_config;
+    driver_config.set_controller_model("RCU");
+    driver_config.set_robot_model("READY-5.0");
+
+    // io_configurationuration
+    auto &io_configuration = *driver_config.mutable_io_configuration();
```
Comment (mechterp10) [inline src/SampleDriver.cpp]: Do we really need to have them create an I/O signal to start? I guess the only downside is that you will get an I/O mismatch warning down the road.

**Review by mechterp10 — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (mechterp10) [reply CMakeLists.txt]: Far enough, lol

Comment (adamcalhoon) [reply CMakeLists.txt]: I'll remove it and see if it still compiles/runs without it.

File: src/SampleDriver.cpp (hunk 5 above)
Comment (adamcalhoon) [reply src/SampleDriver.cpp]: No, I don't think it's required to start up. Copied from comau. I'll remove it.

File: src/SampleDriver.cpp (hunk 4 above)
Comment (adamcalhoon) [reply src/SampleDriver.cpp]: Addressed

File: src/SampleDriver.cpp (hunk 5 above)
Comment (adamcalhoon) [reply src/SampleDriver.cpp]: Addressed.

File: CMakeLists.txt (hunk 1 above)
Comment (adamcalhoon) [reply CMakeLists.txt]: Addressed.

File: configurations/Sample.json (hunk 2 above)
Comment (adamcalhoon) [reply configurations/Sample.json]: Addressed.

File: configurations/Sample.json [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            "modes": ["DISCONNECTED", "TEACH", "TEACH (DISABLED)", "RUN", "TEST"],
+            "fallback": "Refer to the OEM pendant for more information on the active warnings and errors.",
+            "notification": [],
+            "warning": [
+            ],
+            "fault": [
+                {
+                    "code": 100000,
+                    "subject": "Robot Controller Disconnected",
+                    "body": "Custom disconnected fault handling message.",
```
Comment (mechterp10) [inline configurations/Sample.json]: What about this?
```suggestion
                    "body": "TODO: Add custom disconnected fault handling message or remove `disconnected-fault-override-code from the controller configuration JSON file .",
```

**Review by mechterp10 — APPROVED** [review]

**Pull request merged**